### PR TITLE
Squeezenet

### PR DESCRIPTION
**Pull request by huningxin**
Fix https://github.com/intel/webml-polyfill/issues/22 and https://github.com/intel/webml-polyfill/issues/23

### CODE REVIEW

File: services/ml/compilation_impl_mac.h [hunk 1]
```diff
@@ -20,6 +20,14 @@
 
 namespace ml {
 
+struct OperandMac : public Operand {
```
Comment (halton) [inline services/ml/compilation_impl_mac.h]: Can I know why use struct instead of class?

File: services/ml/compilation_impl_mac.h [hunk 2]
```diff
@@ -20,6 +20,14 @@
...
+  OperandMac();
+  OperandMac(const OperandMac&);
+  OperandMac(const Operand&);
+  ~OperandMac();
+  uint32_t read_count;
```
Comment (halton) [inline services/ml/compilation_impl_mac.h]: `read_count_`?

File: services/ml/compilation_impl_mac.mm [hunk 3]
```diff
@@ -480,13 +509,65 @@ void API_AVAILABLE(macosx(10.13)) ComputeMPSOffsetForImplictPadding(
...
+  std::vector<uint32_t> outputs = concat.outputs;
+
+  uint32_t axis = getScalarInt32(values_[inputs[inputs.size() - 1]], memory_.get());
+
+  if (axis != 3) {
+    DLOG(ERROR) << "Only axis == 3 is supported";
+    return false;
+  }
+
+  if (@available(macOS 10.13, *)) {
```
Comment (halton) [inline services/ml/compilation_impl_mac.mm]: Is this check works with 10.13 above?

File: services/ml/compilation_impl_mac.mm [hunk 4]
```diff
@@ -480,13 +509,65 @@ void API_AVAILABLE(macosx(10.13)) ComputeMPSOffsetForImplictPadding(
...
+    DLOG(INFO) << "  Concatenation is compiled to no-op";
+    uint32_t concat_output_idx = concat.outputs[0];
+    uint32_t channelOffset = 0;
+    for (size_t i = 0; i < inputs.size() - 1; ++i) {
+      uint32_t concat_input_idx = inputs[i];
+      for (size_t j = 0; j < operations_.size(); ++j) {
+        OperationMac& operation = operations_[j];
+        if (operation.outputs[0] == concat_input_idx) {
+          DLOG(INFO) << "  Rewrite op " << j << " type " << operation.type <<
+            " output from " << operation.outputs[0] << " to " << concat_output_idx;
```
Comment (halton) [inline services/ml/compilation_impl_mac.mm]: nit, 4-space indent

File: services/ml/compilation_impl_mac.mm [hunk 5]
```diff
@@ -480,13 +509,65 @@ void API_AVAILABLE(macosx(10.13)) ComputeMPSOffsetForImplictPadding(
...
+          operation.outputs[0] = concat_output_idx;
+          MPSCNNKernel* kernel = operation.mpscnn_kernel.get();
+          if (!kernel) {
+            DLOG(ERROR) << "MPSKernel of operation " << j << " type " << operation.type <<
+              " is not found";
```
Comment (halton) [inline services/ml/compilation_impl_mac.mm]: ditto

File: services/ml/compilation_impl_mac.h (hunk 2 above)
Comment (huningxin) [reply services/ml/compilation_impl_mac.h]: public member

File: services/ml/compilation_impl_mac.h (hunk 1 above)
Comment (huningxin) [reply services/ml/compilation_impl_mac.h]: my intent is to simple struct for quick prototyping. we can refactory that later. what do you think?

File: services/ml/compilation_impl_mac.mm (hunk 4 above)
Comment (huningxin) [reply services/ml/compilation_impl_mac.mm]: will do that.

File: services/ml/compilation_impl_mac.mm (hunk 5 above)
Comment (huningxin) [reply services/ml/compilation_impl_mac.mm]: acked

File: services/ml/compilation_impl_mac.mm (hunk 3 above)
Comment (huningxin) [reply services/ml/compilation_impl_mac.mm]: yes.

File: services/ml/compilation_impl_mac.h (hunk 1 above)
Comment (halton) [reply services/ml/compilation_impl_mac.h]: Make sense.

**Comment by halton** [comment]
LGTM and merge.

**Pull request merged**